app/controllers/tools_controller.rb

```diff
@@ -4,23 +4,31 @@ class ToolsController < ApplicationController
   # GET /tools
   # GET /tools.json
   def index
-    unless ( params[:organization_id].blank? )
+    unless params[:organization_id].blank?
       @organization = Organization.find(params[:organization_id])
       @tools = Tool.checked_out_by_organization(@organization)
     end
 
-    if (params[:type].blank?)
+    # Filter by tools
+    if params[:type_filter].present?
+      if params[:type_filter].strip == 'all_tools'
+        @tools = Tool.all
+        @title = "Tools (hardhats/radios included)"
+      else
+        @tool_type = ToolType.find(params[:type_filter])
+        @title = @tool_type.name.pluralize
+        @tools = Tool.by_type(@tool_type)
+        @waitlist = @tool_type.tool_waitlists.by_wait_start_time
+        @num_available = @tools.size - @tools.checked_out.size
+      end
+    else
+      @tools = Tool.just_tools
       @title = "Tools"
-      @tools = @tools.just_tools
-    elsif (params[:type] == 'hardhats')
-      @title = "Hardhats"
-      @tools = @tools.hardhats
-    elsif (params[:type] == 'radios')
-      @title = "Radios"
-      @tools = @tools.radios
-    elsif (params[:type] == 'out')
-      @title = "Checked Out"
-      @tools = @tools.just_tools.checked_out
+    end
+
+    # Fitler by inventory status
+    if params[:inventory_filter].present? && params[:inventory_filter].strip == 'checked_out'
```

Comment:
am not a fan of hardcoded strings D: could we put that in the model or somewhere else?